### DISCUSSION TITLE
persistent, key-value storage life cycle

### DISCUSSION THREAD

**Discussion by kanso-git** [react-native-async-storage/async-storage#806] [Q&A]
I would like to know in which cases the stored data will be cleaned or deleted by the iOS, Android respectively .

Real case scenario:
Imagine I have released a new version of my app that uses already the async-storage to save the user profile 
What will happen to the saved profile on the device after releasing a new version of the app ( app v1.0 -> app v1.1)

I'm sure that the data will be kept when updating the app, but i would to be more sure that since this is the first time using async-storage

Many thanks

**Comment by tido64** [comment]
AsyncStorage doesn't add/remove values unless you tell it to.

The operating system, on the other hand, may delete the store if the app is deleted. It may also restore it if it was synced when reinstalling the app.

**Reply by kanso-git** [reply]
your answer is completely out of the scope of my question:
all I wanted is a confirmation from someone already used async-storage, that the stored data will NOT be removed on app update.

I know that when deleting an app everything related to the app will be deleted , but here the case is different

**Reply by tido64** [reply]
The operating system does not delete your storage when updating an app. If that was the case, you'd be constantly logged out of all your apps all the time.
